hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchFieldsIntegrationTest.kt

```diff
@@ -0,0 +1,329 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch
+
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+import org.springframework.test.web.reactive.server.WebTestClient
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.BodyPartDetail
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.CurrentIncentive
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.IncentiveLevel
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.PrisonerAlert
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.PrisonerAlias
+import uk.gov.justice.digital.hmpps.prisonersearch.search.AbstractSearchIntegrationTest
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.AttributeSearchRequest
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.requestdsl.EQ
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.requestdsl.IS
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.requestdsl.LT
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.requestdsl.RequestDsl
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+/**
+ * We have other tests for different field types and for query structure, but we also need to check that every single
+ * field on the prisoner record is searchable.
+ */
+class AttributeSearchFieldsIntegrationTest : AbstractSearchIntegrationTest() {
```

Comment:
It was frustrating to realise when I started testing that some fields just didn't work at all. So I decided we should test every searchable field somewhere.